src/Tools/dotnet-monitor/Extensibility/ExtensionDeclaration.cs

```diff
@@ -23,6 +23,11 @@ internal class ExtensionDeclaration
         /// </summary>
         public string Program { get; set; }
 
+        /// <summary>
+        /// This is the name that users specify in configuration to refer to the extension.
+        /// </summary>
+        public string DisplayName { get; set; }
```

Comment:
`DisplayName` is not a good name for this. A property named "DisplayName" is typically only used for presenting to an end user with a human readable name, not for identification purposes. I would just class this `Name` for now.

I think we'll have to revisit the format of the extension.json file because it feels like it wants to be partly generic multi-extension manifest and partly specific for egress extensions.